src/protocols/unix_sock/server.rs

```diff
@@ -50,9 +50,14 @@ impl Stream for UnixListenerStream {
 pub async fn run_server(socket_path: &Path) -> Result<UnixListenerStream, anyhow::Error> {
     info!("Starting Unix socket server listening cnx on {:?}", socket_path);
 
-    let path_to_delete = !socket_path.exists();
+    if socket_path.exists() {
```

Comment:
If the unix socket path, already exist we should not delete it. As it is not owned by wstunnel.